### ISSUE TITLE
Not returning the children of path

### ISSUE THREAD

**Issue by dsnowdon** [firebase/polymerfire#87]
### Description

When making a query to a path, for example:
`/inputs/key/`

I expect all the children for the particular path to be returned, the data i'm querying is:

```
-inputs
     -KNC_Z6RkvCAa34qDAX_
         -KNDhXuvg9znsiuNncYV
         -KNDhYwK_wP8K318bntx
         -KNDha0rpnkyRVMWFgHr
         -KNC_Z6RkvCadsfsfsdaf_: 
     - test
        - some data
```

Expected return using the path above: 

```
         -KNDhYwK_wP8K318bntx
         -KNDha0rpnkyRVMWFgHr
         -KNC_Z6RkvCadsfsfsdaf_: 

```

but the actual result is

```

     -KNC_Z6RkvCAa34qDAX_
         -KNDhXuvg9znsiuNncYV
         -KNDhYwK_wP8K318bntx
         -KNDha0rpnkyRVMWFgHr
         -KNC_Z6RkvCadsfsfsdaf_: 
     - test
        - some data
```

Im not sure if it's ment to be like this. And its easily to work around, simply wrap in another child like `inputs/key/data/`

But I would like some clarification.

The query i'm using is
```
<firebase-query
        id="query"
        path="/inputs/{{data.key}}"
        data="{{inputs}}">
      </firebase-query>
```

Its the same output with firebase-document aswell.

### More Info
"polymerfire": "firebase/polymerfire#^0.9.4",
Getting the same result on Chrome and Firefox

A observer watching the data _inputs_, returns:
```
Object {}
[]
Object {-KNC_Z6RkvCAa34qDAX_: Object, -KNC_Z6RkvCadsfsfsdaf_: "test"}

```

**Comment by mbleigh** [comment]
What happens if you remove the trailing slash from your path? Do you have an actual code sample with expected and actual data returned?

**Comment by dsnowdon** [comment]
@mbleigh Thanks for the quick reply, i've added the element declaration i'm using. If you would like more just ask but I didn't think the layout code was relevant.

**Comment by mbleigh** [comment]
Is `data.key` for sure loading? If you hardcode in `/inputs/<THE_SPECIFIC_KEY_YOU_WANT>` does anything change? This seems like `firebase-query` is loading one path level above what you want.

In general, in a structure of `/a/b/{c, d, e, f}`, a `firebase-query` at `/a/b` should load as an array of `[c,d,e,f]`.

**Comment by dsnowdon** [comment]
What do you mean for loading? an app-route element returns the data object

Hardcoding the Id -KNC_Z6RkvCAa34qDAX_ doesn't change anything.

**Comment by ThatBalls** [comment]
I have an almost identical issue except I have a step in the middle where the correct values are being returned.

`    <app-route
        route="{{route}}"
        pattern="/game/:gameKey"
        data="{{routeData}}"></app-route>

    <firebase-document
      path="/characters/{{routeData.gameKey}}"
      data="{{characters}}">
    </firebase-document>`

and this is the console log observing changes:

New Characters: Object {}
New Characters: Object {-KN-LKzWHROWn3YFNCl2: Object, -KN-S_JsgdDwed1FnSoX: Object, -KN-SdvHKSLiEl4PORq8: Object, -KN-T4TSPBGHAr3DDFCo: Object, -KN-TRK9ijb06j0At42C: Object…}
New Characters: Object {-KL3tVPZSIt22wUBRgKq: Object}

the database structure looks like this:

```
characters {
    -KL3tVPZSIt22wUBRgKq {
        -KN-LKzWHROWn3YFNCl2
        -KN-S_JsgdDwed1FnSoX
        -KN-SdvHKSLiEl4PORq8
        -KN-T4TSPBGHAr3DDFCo
        -KN-TRK9ijb06j0At42C
    }
}
```

When I put the hard value directly into the firebase-document instead of getting it from the URL, I get the correct values. Also, if I initialize the component, change the gameKey, and change it back I get the correct values.

It appears to have something to do with initialization and gameKey being an empty string at some point. I'm rather new to Polymer so I'm not sure exactly how to solve this issue.

**Comment by sherif-gamal** [comment]
I was facing the same issue with firebase-document, my element was declared like this

```
 <firebase-document
      path="/users/[[user.uid]]"
      data="{{reminders}}">
</firebase-document>
```

and I found that `__computeRef` gets called with '/users/' first before getting called with the correct user id. After several hours of debugging I found that the reason was that `user` starts undefined until the authentication is complete, I solved the problem by making sure that `user` is not undefined before displaying my element.

**Comment by mbleigh** [comment]
I think #95 may address this issue.

**Comment by mbleigh** [comment]
Hopefully fixed via #95
